contracts/DePayPaymentsV1.sol

```diff
@@ -91,7 +101,7 @@ contract DePayPaymentsV1 is Ownable {
         _pay(payable(addresses[addresses.length-1]), path[path.length-1], amounts[1]);
       } else {
         require(_isApproved(plugins[i]), 'DePay: Plugin not approved!');
-        address plugin = approvedPlugins[plugins[i]];
+        address plugin = configuration.approvedPlugins(plugins[i]);
         (bool success, bytes memory returnData) = plugin.delegatecall(abi.encodeWithSelector(
```

Comment:
We need to check, does plugin able to change `configuration`'s address?.

Reply:
From [the docs](https://solidity.readthedocs.io/en/v0.6.5/contracts.html#immutable):

> Variables declared as immutable are a bit less restricted than those declared as constant: Immutable variables can be assigned an arbitrary value in the constructor of the contract or at the point of their declaration. They cannot be read during construction time and can only be assigned once.

As the configuration address is declared immutable [here](https://github.com/DePayFi/depay-ethereum-payments/pull/11/files#diff-e5ee02f6e184c47c6cb1a9638b14ea78838e3ab3a42da101de3e92e4ac136490R19) it can only be set in the constructor (which happens [here](https://github.com/DePayFi/depay-ethereum-payments/pull/11/files#diff-e5ee02f6e184c47c6cb1a9638b14ea78838e3ab3a42da101de3e92e4ac136490R27)) but not in any other execution of this smart contracts code or any plugin.